bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/IStructuredSelection.java

```diff
@@ -15,27 +15,92 @@
 
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 /**
  * A selection containing elements.
  */
 @SuppressWarnings("rawtypes")
 public interface IStructuredSelection extends ISelection, Iterable {
 	/**
-	 * Returns the first element in this selection, or <code>null</code>
-	 * if the selection is empty.
+	 * Returns the first element in this selection, or <code>null</code> if the
+	 * selection is empty.
 	 *
 	 * @return an element, or <code>null</code> if none
 	 */
-	public Object getFirstElement();
+	default Object getFirstElement() {
+		return stream().findFirst().orElse(null);
+	}
+
+	/**
+	 * Returns an optional describing the first element of the requested type or an
+	 * empty optional if the selection is empty or does not contain any compatible
+	 * element in regard to the specified type
+	 *
+	 * @param type the expected type
+	 * @param <T>  the desired type
+	 * @return an optional describing the result
+	 * @since 3.32
+	 */
+	default <T> Optional<T> getFirstElementOf(Class<T> type) {
```

Comment:
Just `getFirst(Class)` would probably make it more natural to read:
`selection.getFirst(String.class)`